lib/Dialect/FIRRTL/Transforms/Dedup.cpp

```diff
@@ -1399,7 +1399,8 @@ void fixupAllModules(InstanceGraph &instanceGraph) {
         continue;
       ImplicitLocOpBuilder builder(inst.getLoc(), inst->getContext());
       builder.setInsertionPointAfter(inst);
-      for (unsigned i = 0, e = getNumPorts(module); i < e; ++i) {
+      auto e = getNumPorts(module);
+      for (unsigned i = 0; i < e; ++i) {
```

Comment:
![image](https://github.com/llvm/circt/assets/1018530/9f57d767-54a3-4157-8b75-1b7f7f46c921)

I'd be surprised if this is actually fixing it and just permuting things enough that it works. 🤔 Anyway, go for it and see where we get. Thanks for filing this.

Reply:
I said pretty much the same thing over in #6513.